bitflags.h

```diff
@@ -33,59 +35,136 @@ namespace gcpp {
 	//----------------------------------------------------------------------------
 
 	class bitflags {
+		using unit = unsigned int;
+		static_assert(std::is_unsigned<unit>::value, "unit must be an unsigned integral type.");
+
+		std::unique_ptr<unit[]> bits;
 		const int size;
-		std::vector<byte> bits;
+
+		static constexpr std::size_t bits_per_unit = sizeof(unit) * CHAR_BIT;
+
+		//  Return a unit with all bits set if "set" is true, or all bits cleared otherwise.
+		//
+		static constexpr unit all_bits(bool set) {
+			return set ? ~unit(0) : unit(0);
+		}
+
+		//  Return a mask that will select the bit at position from its unit
+		//
+		static unit bit_mask(int at) {
+			Expects(0 <= at && "position must be non-negative");
+			return unit(1) << (at % bits_per_unit);
+		}
+
+		//  Return the number of units needed to represent a number of bits
+		//
+		static int unit_count(int bit_count) {
+			Expects(0 <= bit_count && "bit_count must be non-negative");
+			return (bit_count + bits_per_unit - 1) / bits_per_unit;
+		}
+
+		//  Get the unit that contains the bit at position
+		//
+		unit& bit_unit(int at) {
+			Expects(0 <= at && "position must be non-negative");
+			return bits[at / bits_per_unit];
+		}
+		const unit& bit_unit(int at) const {
+			Expects(0 <= at && "position must be non-negative");
+			return bits[at / bits_per_unit];
+		}
 
 	public:
-		bitflags(int bits, bool value)
-			: size{ bits }
-			, bits(1 + size / sizeof(byte), value ? byte(0xFF) : byte(0x00))
-		{ 
-			Expects(bits > 0 && "#bits must be positive");
+		bitflags(int nbits, bool value)
+			: size{ nbits }
+		{
+			Expects(nbits >= 0 && "#bits must be non-negative");
+			bits = std::make_unique<unit[]>(unit_count(nbits));
+			if (value) {
+				set_all(true);
```

Comment:
Yes. I couldn't remember either, had to look it up, and was surprised to discover this overload of `make_unique` in C++14 - I would have sworn it was added in 17.